runtime/src/xcm_config.rs

```diff
@@ -0,0 +1,259 @@
+	sp_std::marker::PhantomData<(RuntimeCall, AllowedOrigin)>,
+);
+impl<RuntimeCall: Decode, AllowedOrigin: Contains<MultiLocation>> ShouldExecute
+	for AllowUnpaidTransactsFrom<RuntimeCall, AllowedOrigin>
+{
+	fn should_execute<Call>(
+		origin: &MultiLocation,
+		instructions: &mut [Instruction<Call>],
+		max_weight: Weight,
+		_properties: &mut xcm_executor::traits::Properties,
+	) -> Result<(), ProcessMessageError> {
+		log::trace!(
+			target: "xcm::barriers",
+			"AllowUnpaidTransactsFrom origin: {:?}, instructions: {:?}, max_weight: {:?}, properties: {:?}",
+			origin, instructions, max_weight, _properties,
+		);
+
+		// we only allow from configured origins
+		ensure!(AllowedOrigin::contains(origin), ProcessMessageError::Unsupported);
+
+		// we expect an XCM program with single `Transact` call
+		instructions
+			.matcher()
+			.assert_remaining_insts(1)?
+			.match_next_inst(|inst| match inst {
+				Transact { origin_kind: OriginKind::Superuser, .. } => {
+					// we allow all calls through the bridge
+					Ok(())
+				},
+				_ => Err(ProcessMessageError::BadFormat),
+			})?;
+
+		Ok(())
+	}
+}
+
+/// The means that we convert an XCM origin `MultiLocation` into the runtime's `Origin` type for
+/// local dispatch. This is a conversion function from an `OriginKind` type along with the
+/// `MultiLocation` value and returns an `Origin` value or an error.
+type LocalOriginConverter = (
+	// Currently we only accept XCM messages from Kawabunga and the origin for such messages
+	// is local root.
+	KawabungaParachainAsRoot,
+);
+
+// TODO [bridge]: configure to use bridge blob hauler here
+/// Only bridged destination is supported.
+pub type XcmRouter = ();
+
+/// The barriers one of which must be passed for an XCM message to be executed.
+pub type Barrier = TrailingSetTopicAsId<
+	WithComputedOrigin<
+		// We only allow unpaid execution from the Kawabunga parachain.
+		AllowUnpaidTransactsFrom<RuntimeCall, OnlyKawabungaLocation>,
+		UniversalLocation,
+		ConstU32<2>,
+	>,
+>;
+
+/// XCM executor configuration.
+pub struct XcmConfig;
+
+impl xcm_executor::Config for XcmConfig {
+	type RuntimeCall = RuntimeCall;
+	type XcmSender = XcmRouter;
+	type AssetTransactor = ();
+	type OriginConverter = LocalOriginConverter;
+	type IsReserve = ();
+	type IsTeleporter = ();
+	type UniversalLocation = UniversalLocation;
+	type Barrier = Barrier;
+	// TODO [bridge]: is it ok to use `FixedWeightBounds` here? We don't have the `pallet-xcm`
+	// and IIUC can't use XCM benchmarks because of that?
+	type Weigher = FixedWeightBounds<BaseXcmWeight, RuntimeCall, MaxInstructions>;
+	type Trader = NoopTrader;
+	type ResponseHandler = ();
+	type AssetTrap = ();
+	type AssetLocker = ();
+	type AssetExchanger = ();
+	type AssetClaims = ();
+	type SubscriptionService = ();
+	type PalletInstancesInfo = AllPalletsWithSystem;
+	type MaxAssetsIntoHolding = ConstU32<0>;
+	type FeeManager = ();
+	type MessageExporter = ();
+	type UniversalAliases = Nothing;
```

Comment:
@svyatonik 
Barrier passes, but real execution will fail because of `type UniversalAliases = Nothing;`
```suggestion
	type UniversalAliases = UniversalAliases;
```